node/core/pvf/tests/it/worker_common.rs

```diff
@@ -46,3 +58,72 @@ async fn should_connect() {
 	.await
 	.unwrap();
 }
+
+#[test]
+fn get_condvar_should_be_pending() {
+	let condvar = get_condvar();
+	let outcome = *condvar.0.lock().unwrap();
+	assert!(outcome.is_pending());
+}
+
+#[test]
+fn wait_for_threads_with_timeout_return_none_on_time_out() {
+	let condvar = Arc::new((Mutex::new(WaitOutcome::Pending), Condvar::new()));
+	let outcome = wait_for_threads_with_timeout(&condvar, Duration::new(1, 0));
+	assert!(matches!(outcome, None));
+}
+
+#[test]
+fn wait_for_threads_with_timeout_returns_outcome() {
+	let condvar = Arc::new((Mutex::new(WaitOutcome::Pending), Condvar::new()));
+	let condvar2 = Arc::clone(&condvar);
+	std::thread::spawn(move || {
+		let (lock, cvar) = &*condvar2;
+		let mut outcome = lock.lock().unwrap();
+		*outcome = WaitOutcome::Finished;
+		cvar.notify_one();
+	});
+	let outcome = wait_for_threads_with_timeout(&condvar, Duration::from_secs(2));
+	assert!(matches!(outcome.unwrap(), WaitOutcome::Finished));
+}
+
+#[test]
+fn spawn_worker_thread_should_notify_on_done() {
+	let condvar = Arc::new((Mutex::new(WaitOutcome::Pending), Condvar::new()));
+	let response =
+		spawn_worker_thread("thread", move || 2, Arc::clone(&condvar), WaitOutcome::TimedOut);
+	let (lock, _) = &*condvar;
+	let r = response.expect("").join().unwrap();
```

Comment:
I think this can cause a deadlock, we get the lock on the condvar first so the thread may not be able to get the lock to signal that it's finished. We should join on the thread first. Also, can use `.unwrap()` instead of `.expect("")`. :)

```suggestion
	let r = response.unwrap().join().unwrap();
	let (lock, _) = &*condvar;
```

Same applies to `spawn_worker_thread_should_not_notify` below.